packages/core/src/routes/interaction/additional.ts

```diff
@@ -128,10 +139,35 @@ export default function additionalRoutes<T extends IRouterParamContext>(
         true
       );
 
-      ctx.body = {
-        secret,
-        secretQrCode: await qrcode.toDataURL(`otpauth://totp/?secret=${secret}`),
-      };
+      if (isRegisterInteractionResult(profileVerifiedInteraction)) {
+        const {
+          username = null,
+          primaryEmail = null,
+          primaryPhone = null,
+          name = null,
```

Comment:
Looks like these values could remain undefined.

Reply:
It won't.

<img width="264" alt="image" src="https://github.com/logto-io/logto/assets/5717882/5e471998-4f18-41f2-ab26-e3401970889a">